component-framework/TS_PropertySetTableControl/TSPropertySetTableControl/index.ts

```diff
@@ -0,0 +1,355 @@
+	 */
+	private getColumnWidthDistribution(context: ComponentFramework.Context<IInputs>, columnsOnView: DataSetInterfaces.Column[]): string[] {
+
+		let widthDistribution: string[] = [];
+
+		// Considering need to remove border & padding length
+		let totalWidth: number = context.mode.allocatedWidth;
+		let widthSum = 0;
+
+		columnsOnView.forEach(function (columnItem) {
+			widthSum += columnItem.visualSizeFactor;
+		});
+
+		let remainWidth: number = totalWidth;
+
+		columnsOnView.forEach(function (item, index) {
+			let widthPerCell = "";
+			if (index !== columnsOnView.length - 1) {
+				let cellWidth = Math.round((item.visualSizeFactor / widthSum) * totalWidth);
+				remainWidth = remainWidth - cellWidth;
+				widthPerCell = cellWidth + "px";
+			}
+			else {
+				widthPerCell = remainWidth + "px";
+			}
+			widthDistribution.push(widthPerCell);
+		});
+
+		return widthDistribution;
+
+	}
+
+	private createTableHeader(columnsOnView: DataSetInterfaces.Column[], widthDistribution: string[]): HTMLTableSectionElement {
+
+		let tableHeader: HTMLTableSectionElement = document.createElement("thead");
+		let tableHeaderRow: HTMLTableRowElement = document.createElement("tr");
+		tableHeaderRow.classList.add("SimpleTable_TableRow_Style");
+		const _this = this;
+		columnsOnView.forEach(function (columnItem, index) {
+			let tableHeaderCell = document.createElement("th");
+			tableHeaderCell.classList.add("SimpleTable_TableHeader_Style");
+			let innerDiv = document.createElement("div");
+			innerDiv.classList.add("SimpleTable_TableCellInnerDiv_Style");
+			innerDiv.style.maxWidth = widthDistribution[index];
+			let columnDisplayName: string;
+			if (columnItem.order < 0) {
+				columnDisplayName = columnItem.displayName + "(propertySet)";
+			} else {
+				columnDisplayName = columnItem.displayName;
+			}
+			innerDiv.innerText = columnDisplayName;
+
+			tableHeaderCell.appendChild(innerDiv);
+			tableHeaderRow.appendChild(tableHeaderCell);
+		});
+
+		tableHeader.appendChild(tableHeaderRow);
+		return tableHeader;
+	}
+
+	private createSearchBar(context: ComponentFramework.Context<IInputs>) {
+		let container = document.createElement("div");
+		let input = document.createElement("input");
+		input.id = "searchBar";
+		let button = document.createElement("button");
+		button.innerHTML = "Search"
+		button.addEventListener("click", (() => {
+			let conditionsArray: DataSetInterfaces.ConditionExpression[] = [];
+			let searchString = input.value;
+			for (let i = 0; i < this.columns.length; i++) {
+				const column = this.columns[i];
+				if (!column.isHidden && column.dataType === "SingleLine.Text") {
+					const condition: DataSetInterfaces.ConditionExpression = {
+						attributeName: column.alias,
+						conditionOperator: 6,
+						value: searchString,
+					};
+					conditionsArray.push(condition);
+				}
+			}
+			this.contextObj.parameters.sampleDataSet.filtering.setFilter({
+				conditions: conditionsArray,
+				filterOperator: 1,
```

Comment:
what is this code doing ?

Reply:
This is to wrap the conditions under an OR operator